deployment-guide.md

```diff
@@ -20,44 +20,27 @@ To be able to do that, your development environment shall already have Docker da
 Note, that the project itself depends on Golang 1.10+ and glide being available, but we packaged these dependencies into an automatically created builder container, so you don't have to worry about them!
 ### Building the binaries
 
-It is actually as easy as go get-ting the repository from GitHub, and executing the build_danm.sh script from the root of the project!
+It is actually as easy as cloning the repository from GitHub, and executing the build_danm.sh script from the root of the project!
+
 ```
-go get -d github.com/nokia/danm
-cd $GOPATH/src/github.com/nokia/danm
+git clone github.com/nokia/danm
+cd danm
 ./build_danm.sh
 ```
-This will first build the Alpine based builder container, mount the $GOPATH/src and the $GOPATH/bin directory into it, and invoke the necessary script to build all binaries inside the container.
-The builder container destroys itself once its purpose has been fulfilled.
 
-The result will be 6, statically linked binaries put into your $GOPATH/bin directory.
+The result will four container images:
 
-- danm
+- `danm-cni-plugins`: This image contains the core CNI plugins (`danm`, `fakeipam`) as well as the `macvtap`
+  plugin. It will be deployed as a DaemonSet that puts these binaries in place in each Kubernetes node.
 
-- fakeipam
+- `netwatcher`: This image will be used by the `netwatcher` DaemonSet
```

Comment:
what we did in our product is that we ultimately created a "hyperdanm" container image, with all three binaries
I think it would be the easiest way to go

Reply:
Could do, but how strong do you feel about that? It would reduce the size of the Dockerfile. it would marginally reduce the total consumed space for container images, but I have to admit I like the idea of having three separate images for the separate pods. Keeps things tidy somehow; each pod contains only the binary that it actually needs.

For the user, it should not make a difference anyway -- their `kubectl apply` look all the same :)

Reply:
How about supporting both and avoid tying the users' hands?

Reply:
Are you suggesting to build both, and let the user chose which to install?

Or to offer two versions of the build script (or one script with conditionals), and two paths of documentation?

Reply:
I suggest building both the `single binary per image` and the `multi binary per image` method, because as @Levovar mentioned in our product we prefer the `multi binary image` method, butI see other users (including You also?🌝) would prefer single binary images.
> Is this something that could potentially be done in a future enhancement?

Yes, of course. If you are not interested in the multi binary image method, just leave it to me, I will add it later after your PR is merged.

Reply:
I'm happy to look at it over the next few days - but if you guys can live with the current version, I'd probably prefer to merge this as-is, and then work on it as an incremental change that adds the option to build a second form factor.

Reply:
yeah just an idea, as you can see I feel so strongly about it I also haven't brought it over to upstream in the last 6 months :)